dashboard/src/components/Config/AppRepoList/AppRepoButton.v2.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from "react";
+
+import actions from "actions";
+import { CdsButton, CdsIcon } from "components/Clarity/clarity";
+import Alert from "components/js/Alert";
+import Modal from "components/js/Modal/Modal";
+import { useDispatch, useSelector } from "react-redux";
+import { Action } from "redux";
+import { ThunkDispatch } from "redux-thunk";
+import { IAppRepository, ISecret, IStoreState } from "../../../shared/types";
+import "./AppRepo.css";
+import { AppRepoForm } from "./AppRepoForm.v2";
+
+interface IAppRepoAddButtonProps {
+  namespace: string;
+  kubeappsNamespace: string;
+  text?: string;
+  primary?: boolean;
+  repo?: IAppRepository;
+  secret?: ISecret;
+}
+
+export function AppRepoAddButton({
+  text,
+  primary,
+  namespace,
+  kubeappsNamespace,
+  repo,
+  secret,
+}: IAppRepoAddButtonProps) {
+  const dispatch: ThunkDispatch<IStoreState, null, Action> = useDispatch();
+  const [modalIsOpen, setModalOpen] = useState(false);
+  const openModal = () => setModalOpen(true);
+  const closeModal = () => setModalOpen(false);
+  const { errors } = useSelector((state: IStoreState) => state.repos);
+  // Primary by default
+  const isPrimary = primary !== false;
```

Comment:
This looks like it's handling the possibility of an `undefined` for primary as it's optional. In which case, why not just define the default above in the function args as `primary = true, ...` (see https://stackoverflow.com/a/32596200 )